horace_core/sqw/@Experiment/private/build_from_old_headers_.m

```diff
@@ -40,45 +65,91 @@
             class(instr),i)
     end
     
+    % Locally name the sample (again, not liking plural samples, same reason)
     sampl = hdr.samples;
-    %
-    if isstruct(sampl) && isempty(fieldnames(sampl))
-        %TODO: should it be sample factory?, returning samples as function
-        % of inputs. The same as instrument
-        sampl = IX_samp();
-        sampl.alatt = alatt;
-        sampl.angdeg = angdeg;
-    elseif isa(sampl,'IX_samp') && isempty(sampl)
+    
+    % If sampl is a struct
+    if isstruct(sampl)
+        % Sample in header is an empty struct, replace with null sample and
+        % populate with the lattice parameters
+        if isempty(sampl) || isempty(fieldnames(sampl))
+            sampl = IX_null_sample();
         sampl.alatt = alatt;
         sampl.angdeg = angdeg;
+            samples(i) = sampl;
+        % struct may have enough info to make a sample (though this is not
+        % defined yet - call to factory method will probably fail)
     else
-        sampl = IX_samp(sampl);
-        if isempty(sampl)
+            samples(i) = make_sample_from_struct(sampl);
+        end 
+    % Sample is actually a subclass of IX_samp, so keep it but overwrite
+    % the lattice parameters if they were not in the old version
+    elseif isa(sampl,'IX_samp')
+        if isempty(sampl.alatt)
             sampl.alatt = alatt;
+        elseif sampl.alatt ~= alatt
+            warning('HORACE:Experiment:invalid parameter',...
```

Comment:
Does `invalid parameter` want to be underscore separated?

Reply:
thanks